rustls/src/server/tls13.rs

```diff
@@ -1032,7 +1033,7 @@ impl hs::State for ExpectFinished {
         let read_key = self
             .key_schedule
             .client_application_traffic_secret(
-                &self.handshake.hash_at_server_fin,
+                self.handshake.hash_at_server_fin.as_ref().unwrap(),
```

Comment:
It's possible to avoid this unwrap but it requires refactoring some parts of the state machine. I have a PR pending, for after this is merged, to thread this digest through the state machine so that this unwrap goes away. That PR eliminates `HandshakeDetails::hash_at_server_fin` completely.